dags/synapse-datasets-created-7-days.py

```diff
@@ -0,0 +1,173 @@
+"""This script executes a query on Snowflake to retrieve data about entities created in the past 7 days and stores the results in a Synapse table.
+It is scheduled to run every Monday at 00:00 UTC.
+"""
+
+from dataclasses import dataclass
+from datetime import date, datetime
+from typing import List
+
+import synapseclient
+from airflow.decorators import dag, task
+from airflow.models import Variable
+from airflow.models.param import Param
+from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
+from orca.services.synapse import SynapseHook
+from slack_sdk import WebClient
+
+dag_params = {
+    "snowflake_conn_id": Param("SNOWFLAKE_SYSADMIN_PORTAL_RAW_CONN", type="string"),
+    "synapse_conn_id": Param("SYNAPSE_ORCA_SERVICE_ACCOUNT_CONN", type="string"),
+    "current_date": Param(date.today().strftime("%Y-%m-%d"), type="string"),
+}
+
+dag_config = {
+    "schedule_interval": "0 0 * * 1",
+    "start_date": datetime(2025, 2, 1),
+    "catchup": False,
+    "default_args": {
+        "retries": 1,
+    },
+    "tags": ["snowflake"],
+    "params": dag_params,
+}
+
+SYNAPSE_RESULTS_TABLE = "syn64919238"
+
+
+@dataclass
+class EntityCreated:
+    """Dataclass to record
+
+    Attributes:
+    name: The name of this entity.
+    id: The unique immutable ID for this entity.
+    parent_id: parent id of the entity
+    node_type: Type of Node
+    content_type: Content type of folder annotated,
+    created_on: The date this entity was created.
+    created_by: The ID of the user that created this entity.
+    is_public: True to indicate if this entity is public.
+    """
+
+    name: str
+    id: int
+    parent_id: int
+    node_type: str
+    content_type: str
+    created_on: str
+    created_by: str
+    is_public: bool
+
+
+@dag(**dag_config)
+def datasets_or_projects_created_7_days() -> None:
+    """Execute a query on Snowflake and report the results to a Synapse table."""
+
+    @task
+    def get_datasets_projects_created_7_days(**context) -> List[EntityCreated]:
+        """Execute the query on Snowflake and return the results."""
+        snow_hook = SnowflakeHook(context["params"]["snowflake_conn_id"])
+        ctx = snow_hook.get_conn()
+        cs = ctx.cursor()
+        query = f"""
```

Comment:
@BWMac I saw that other queries like [this one](https://github.com/Sage-Bionetworks-Workflows/orca-recipes/blob/main/dags/top-public-synapse-projects-all-time-from-snowflake.py#L65) so I also kept my f string here. Is there any reason that we should use a f string in query? I am not inserting anything in the query.

Reply:
If you aren't inserting anything, then no.

Reply:
In the `push_results_to_synapse_table` function, I already added today's date as 'export date.' Is that sufficient, or do we still need to insert today's date in the query here?

Reply:
No we should insert the date into the query so that we can pull data for dates other than just today. The way this DAG is written, you could run it for a date in the past and then the Synapse table will be displaying today's data but it would look like it is from the `current_date`

Reply:
Thanks for the comment, Brad! I will mark this as resolved now.